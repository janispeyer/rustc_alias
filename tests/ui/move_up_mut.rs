// compile-flags: -Zmir-emit-retag

fn move_up_mut(x: &mut i32, y: &mut i32) -> i32 {
    *x = 42;
    *y = 7;
    return *x;
}

fn main() {
    let mut x = 1;
    let mut y = 2;
    move_up_mut(&mut x, &mut y);
}
