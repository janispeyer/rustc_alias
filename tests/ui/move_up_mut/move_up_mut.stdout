# Analysing main
# Analysing move_up_mut
# MaybeTopOfBorrowStack Analysis
bb0
[ 0] [_1, _2] -> Retag([fn entry] _1)
[ 1] [_1, _2] -> Retag([fn entry] _2)
[ 2] [_1, _2] -> (*_1) = const 42_i32
[ 3] [_1, _2] -> (*_2) = const 7_i32
[ 4] [_1, _2] -> _0 = (*_1)
[ 5] [_1, _2] -> return

# FindImmutabilitySpans Analysis
bb0
[ 0] {} -> Retag([fn entry] _1)
[ 1] {} -> Retag([fn entry] _2)
[ 2] {_1: Span(bb0[2])} -> (*_1) = const 42_i32
[ 3] {_1: Span(bb0[2]), _2: Span(bb0[3])} -> (*_2) = const 7_i32
[ 4] {_1: Span(bb0[2]), _2: Span(bb0[3])} -> _0 = (*_1)
[ 5] {_1: Span(bb0[2]), _2: Span(bb0[3])} -> return


# CFG Before
[
    BasicBlockData {
        statements: [
            Retag([fn entry] _1),
            Retag([fn entry] _2),
            (*_1) = const 42_i32,
            (*_2) = const 7_i32,
            _0 = (*_1),
        ],
        terminator: Some(
            Terminator {
                source_info: SourceInfo {
                    span: $DIR/move_up_mut.rs:7:2: 7:2 (#0),
                    scope: scope[0],
                },
                kind: return,
            },
        ),
        is_cleanup: false,
    },
]
# CFG After
[
    BasicBlockData {
        statements: [
            Retag([fn entry] _1),
            Retag([fn entry] _2),
            _4 = const 42_i32,
            (*_1) = _4,
            _5 = const 7_i32,
            (*_2) = _5,
            _0 = _4,
        ],
        terminator: Some(
            Terminator {
                source_info: SourceInfo {
                    span: $DIR/move_up_mut.rs:7:2: 7:2 (#0),
                    scope: scope[0],
                },
                kind: return,
            },
        ),
        is_cleanup: false,
    },
]

